### PR TITLE
Add cluster asset to launch an openshift cluster

### PR DESCRIPTION
**Pull request by yifan-gu**
- Move some helper function from `installer/pkg/workflow` to `pkg/terraform`
- Create a `cluster` asset that uses the `tfvars` asset, terraform templates and terraform binary to launch a cluster
- Add a `clutser` target in the CLI so users can run the CLI to create a cluster.

### CODE REVIEW

File: pkg/asset/cluster/cluster.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+	var result asset.State
+
+	for _, step := range steps {
+		templateDir, err := terraform.FindStepTemplates(dir, step, tfvars.Platform)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := terraform.Init(c.rootDir, templateDir); err != nil {
```
Comment (abhinavdahiya) [inline pkg/asset/cluster/cluster.go]: nit: `terraform.Apply` might do the init internally?

File: pkg/asset/cluster/cluster.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+			return nil, err
+		}
+
+		stateFile, err := terraform.Apply(c.rootDir, step, templateDir)
+		if err != nil {
+			return nil, err
+		}
+
+		data, err := ioutil.ReadFile(state)
```
Comment (abhinavdahiya) [inline pkg/asset/cluster/cluster.go]: `/s/state/stateFile` ?

File: pkg/terraform/terraform.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	stateFileName := fmt.Sprintf("%s.tfstate", state)
+	defaultArgs := []string{
+		"apply",
+		"-auto-approve",
+		fmt.Sprintf("-state=%s", stateFileName),
+	}
+	extraArgs = append(extraArgs, templateDir)
+	args := append(defaultArgs, extraArgs...)
+
+	return path.Join(clusterDir, stateFileName), terraformExec(clusterDir, args...)
```
Comment (abhinavdahiya) [inline pkg/terraform/terraform.go]: Apply is not executing `terraform`

File: pkg/asset/cluster/cluster.go (hunk 1 above)
Comment (yifan-gu) [reply pkg/asset/cluster/cluster.go]: sure.

Comment (yifan-gu) [reply pkg/asset/cluster/cluster.go]: Hmm, maybe just leave as is for now?

**Comment by jlebon** [comment]
Yeah, I hit that too and opened #316. I worked around it for now by just commenting it out.

**Pull request merged**